### DISCUSSION TITLE
Ability to add List of Flows to Flowbuilder's add function

### DISCUSSION THREAD

**Discussion by opticSquid** [spring-projects/spring-batch#4630] [Q&A]
I was working on a spring batch where I have `n` number of vendors. Vendors are mentioned in the configuration file as a list. Number of vendors can change. For every vendor I needed to do the same steps with some parameter changes like the sql query for the item reader and all other wise everything is same. So, I thought of doing the operation for every vendor in parallel using `SimpleAsycTaskExecutor` as `TaskExecutor`. I was doing something like this:
```java
@Bean
Flow splitFlow(TaskExecutor taskExecutor, Flow flow1, Flow flow2, Flow flow3){
    return new FlowBuilder<SimpleFlow>("splitFlow")
                         .split(taskExecutor)
                         .add(flow1, flow2, flow3)
                         .build();
}
```
Here flow1, flow2, flow3 are flows that contain same steps with some vendor specific changes. What I want is to define a dynamic number of flows as a list of Flow objects. and pass the list to the add function. In this way I don't have to define a separate flow every time a vendor is added or removed. I would just iterate through the vendor list and create a Flow object for each of them. 

The current implementation of the add function is like this:
```java
public FlowBuilder<Q> add(Flow... flows){
Collection<Flow> list = new ArrayList<>(Arrays.asList(flows));
//.... rest of the code
}
```
You see in the function it is being treated as a list only so why not let me pass the arguments as `List<Flow>` only. We can create an additional implementation of  `add()`  function that accepts `List<Flow>` type argument. This will help create job with dynamic number of parallel Flows and I don't have to do any code level change every time some change occur in vendor list.

If the add function's implementation would be like this:
```java
public FlowBuilder<Q> add(List<Flow> flows){
// we can then hopefully skip this line entirely
Collection<Flow> list = new ArrayList<>(Arrays.asList(flows));
//.... rest of the code
}
```
I can in my code do like this:
```java
@Bean
Flow splitFlow(TaskExecutor taskExecutor, Flow flow1, Flow flow2, Flow flow3){
   List<Flow> flows =  new ArrayList<>();
   // iterate thorugh vendor list and add Flow objects to the flows list
    return new FlowBuilder<SimpleFlow>("splitFlow")
                         .split(taskExecutor)
                         .add(flows)
                         .build();
}
```
I may be entirely off in my approach so, I need suggestions that is why I'm opening this in discussion not in issues. I need suggestions regarding how do I approach this problem.

**Comment by fmbenhassine** [comment]
What about preparing the builder then calling `add` in a loop? Something like the following example (which adds steps dynamically): https://stackoverflow.com/questions/54853908/how-to-create-dynamic-steps-in-spring-batch

**Comment by opticSquid** [comment]
Ok currently I have gone for an alternative work around solution but I will surely try this one too..
